src/terms/xc.jl

```diff
@@ -82,46 +85,49 @@ end
     tσ = libxc_spinindex_σ
 
     # Potential contributions Vρ -2 ∇⋅(Vσ ∇ρ) + ΔVl
-    potential = zero(ρ)
-    @views for s in 1:n_spin
-        potential[:, :, :, s] .+= terms.vrho[s, :, :, :]
-        if haskey(terms, :vsigma) && any(x -> abs(x) > term.potential_threshold, terms.vsigma)
-            # Need gradient correction
-            # TODO Drop do-block syntax here?
-            potential[:, :, :, s] .+= -2divergence_real(basis) do α
-                # Extra factor (1/2) for s != t is needed because libxc only keeps σ_{αβ}
-                # in the energy expression. See comment block below on spin-polarised XC.
-                sum((s == t ? one(T) : one(T)/2)
-                    .* terms.vsigma[tσ(s, t), :, :, :] .* density.∇ρ_real[t, :, :, :, α]
-                    for t in 1:n_spin)
-            end
-        end
-        if haskey(terms, :vlapl) && any(x -> abs(x) > term.potential_threshold, terms.vlapl)
-            @warn "Meta-GGAs with a Vlapl term have not yet been thoroughly tested." maxlog=1
-            mG² = [-sum(abs2, G) for G in G_vectors_cart(basis)]
-            Vl_fourier = r_to_G(basis, terms.vlapl[s, :, :, :])
-            potential[:, :, :, s] .+= G_to_r(basis, mG² .* Vl_fourier)  # ΔVl
-        end
-    end
+    potential = vrho #zero(ρ)
+    #@views for s in 1:n_spin
```

Comment:
Well this is block of commented code is clearly annoying and my PR will also not solve this issue. We should discuss ways to change primal here to make this more easily differentiable.

Problem is that keeping multiple potentials in memory is quite a bit of a memory overhead. Perhaps the easiest here is to formulate this as a map over `s` returning a `Vector{Array{3}}` and then placing that into the `RealSpaceMultiplication` further below?